### PR TITLE
Add ncbi-amr (antimicrobial resistance) recipe.

### PR DESCRIPTION
**Pull request by slottad**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it. Note that if you are not already a member of the bioconda project (meaning that you can't add this label), please ping `@bioconda/core` so that your PR can be reviewed and merged (please note if you'd like to be added to the bioconda project). Please see https://github.com/bioconda/bioconda-recipes/issues/15332 for more details.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/guidelines.html).
* [x] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [ ] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

File: recipes/ncbi-amr/compile-fix.patch [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+index 1417b54..99a24a4 100644
+--- a/Makefile
++++ b/Makefile
+@@ -27,7 +27,8 @@ CXX=g++
+ COMPILE.cpp= $(CXX) $(CPPFLAGS)  -c
+ 
+ #prefix = /panfs/pan1.be-md.ncbi.nlm.nih.gov/bacterial_pathogens/backup
+-prefix=/usr/local
++#prefix=/usr/local
++prefix=${PREFIX}
```
Comment (epruesse) [inline recipes/ncbi-amr/compile-fix.patch]: You can just write
```
make prefix=$PREFIX
```
to achieve the same. Variables passed like that always override everything in the Makefile.

File: recipes/ncbi-amr/compile-fix.patch [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+ 
+ # soft links are created here to INSTALL_DIR
+ bindir="$(prefix)/bin"
+@@ -71,7 +72,7 @@ clean:
+ 	#rm version.txt
+ 
+ install:
+-	$(INSTALL) -D -t $(INSTALL_DIR) amr_report fasta_check gff_check amrfinder.pl
++	$(INSTALL) -D -t $(bindir) amr_report fasta_check gff_check amrfinder.pl
```
Comment (epruesse) [inline recipes/ncbi-amr/compile-fix.patch]: Same here

File: recipes/ncbi-amr/compile-fix.patch [hunk 3]
```diff
@@ -0,0 +1,44 @@
+diff --git a/Makefile b/Makefile
+index 1417b54..99a24a4 100644
+--- a/Makefile
++++ b/Makefile
+@@ -27,7 +27,8 @@ CXX=g++
+ COMPILE.cpp= $(CXX) $(CPPFLAGS)  -c
```
Comment (epruesse) [inline recipes/ncbi-amr/compile-fix.patch]: Make sure the `Makefile` doesn't set CPPFLAGS to something weird. If it does, use `make CPPFLAGS="$CXXFLAGS ...(additional flags)..."`

Comment (epruesse) [reply recipes/ncbi-amr/compile-fix.patch]: (Technically, CPP is in Make land C-Pre-Processor, while CXX is C++, so CPPFLAGS are things like `-I` and `-D` controlling the preprocessor, and CXXFLAGS are things like `-fopenmp`. But then, lots of Makefiles get this mixed up...)

File: recipes/ncbi-amr/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+build:
+  number: 0
+
+requirements:
+  build:
+    - {{ compiler('cxx') }}
+  host:
+    - coreutils
+  run:
+    - blast
```
Comment (epruesse) [inline recipes/ncbi-amr/meta.yaml]: `blast >2.7` or so? The legacy 2.2 versions can be a bit different, IIRC.

File: recipes/ncbi-amr/compile-fix.patch (hunk 1 above)
Comment (slottad) [reply recipes/ncbi-amr/compile-fix.patch]: Will do, thanks.

File: recipes/ncbi-amr/compile-fix.patch (hunk 2 above)
Comment (slottad) [reply recipes/ncbi-amr/compile-fix.patch]: Yes, will do.

File: recipes/ncbi-amr/meta.yaml (hunk 4 above)
Comment (slottad) [reply recipes/ncbi-amr/meta.yaml]: Good idea.

**Pull request merged**